DragonSlayer.cpp

```diff
@@ -1,21 +1,34 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer(std::string name_, int hitPoints_, int armor_) : Character(hitPoints_, armor_, 4),name(name_)
+{
+    helpfulItems = makeHelpfulItems(2);
+    defensiveItems = makeDefensiveItems(1);
+}
 
-//DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
+        double boostAttack = attackDamage * 10;
+        Character::boostAttackDamage(boostAttack);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
         //note: they should only use the item if the dragon's hitpoints are > 0...
         //note: items are single-use only, so you need to reset it after use.  
         //look in the Character class for how the other item types are reset after use.
+        if ( attackItem != nullptr && dragon->getHP() < 0 )
+        {
+            attackItem->use(this);
+        }
```

Comment:
The attack item is what performs the boost. 
You need to reset it after it is used.
Also, if the dragon's hit points are less than zero, what is the point of using the attack item? 
- [ ] revise